tests/test_query.py

```diff
@@ -150,11 +150,53 @@ def test_get_invalid_age(
 ):
     """Given an invalid age range, returns a 422 status code."""
 
-    async def mock_get(age_min, age_max, sex):
+    async def mock_get(age_min, age_max, sex, image_modal):
         return None
 
     monkeypatch.setattr(crud, "get", mock_get)
     response = test_app.get(
         f"/query/?age_min={invalid_age_min}&age_max={invalid_age_max}"
     )
     assert response.status_code == 422
+
+
+@pytest.mark.parametrize(
+    "valid_image_modal",
+    [
+        "All",
+        "Diffusion weighted",
+        "EEG",
+        "Flow weighted",
+        "T1 weighted",
+        "T2 weighted",
+    ],
+)
+def test_get_valid_image_modal(
+    test_data, test_app, valid_image_modal, monkeypatch
+):
+    "Given a valid image modality, returns a 200 status code and a non-empty list of results."
+
+    async def mock_get(age_min, age_max, sex, image_modal):
+        return test_data
+
+    monkeypatch.setattr(crud, "get", mock_get)
+    response = test_app.get(f"/query/?image_modal={valid_image_modal}")
+    assert response.status_code == 200
+    assert response.json() != []
+
+
+@pytest.mark.parametrize(
+    "invalid_image_modal",
+    ["diffusion weighted", "eeg", "Flow weghed", "T1weighted", "apple"],
```

Comment:
This is a bit picky and I'll defer to @surchs' opinion here, but I would lean towards not having lowercase image modality names as 'invalid' test cases. I think in the case of an isolated API, we would want str params to be case-insensitive (i.e., both "diffusion weighted" and "Diffusion weighted" should be valid inputs). Obviously, since we'll soon have a GUI frontend that presumably has exact options the user can select rather than accepting any str, I don't think we urgently need to implement case-insensitivity right now - but maybe it would still be better conceptually to avoid having them in the codebase as explicit examples of invalid parameters?

Reply:
I was hoping this would come up so thanks for mentioning it. I wasn't sure how restrictive we'd wanna be with the input so I decided to go with the most restrictive and intentionally mentioned those cases in test for demonstration.
I'd be happy to implement case insensitivity if we end up going that way.